autoload/multiple_cursors.vim

```diff
@@ -457,9 +481,27 @@ function! s:CursorManager.update_current() dict
     exec "normal! gv\<Esc>"
     call cur.update_visual_selection(s:get_visual_region(s:pos('.')))
   elseif s:from_mode ==# 'v' || s:from_mode ==# 'V'
+    " Save contents of unnamed register after each operation in Visual mode.
+    " This should be executed after user input is processed, when unnamed
+    " register already contains the text.
+    if s:individual_paste_buffer_enabled()
+      call cur.save_paste_buffer()
+    endif
+
     call cur.remove_visual_selection()
   elseif s:from_mode ==# 'i' && s:to_mode ==# 'n' && self.current_index != self.size() - 1
     normal! h
+  elseif s:from_mode ==# 'n'
+    " Save contents of unnamed register after each supported operation in Normal
+    " mode. Paste buffer should be saved only if we are about to perform a
+    " supported operation that is known to change unnamed register. List of
+    " supported operations is fixed and defined above.
+    " This is ugly and can probably be done more elegantly and reliably.
```

Comment:
Hmm, what about we just checked after every operation on every cursor if the register contents have changed, and if they have, save?